include/SDL3/SDL_audio.h

```diff
@@ -807,13 +807,70 @@ extern DECLSPEC int SDLCALL SDL_SetAudioStreamFrequencyRatio(SDL_AudioStream *st
  *
  * \since This function is available since SDL 3.0.0.
  *
+ * \sa SDL_PutAudioStreamBuffer
  * \sa SDL_ClearAudioStream
  * \sa SDL_FlushAudioStream
  * \sa SDL_GetAudioStreamData
  * \sa SDL_GetAudioStreamQueued
  */
 extern DECLSPEC int SDLCALL SDL_PutAudioStreamData(SDL_AudioStream *stream, const void *buf, int len);
 
+/**
+ * A callback that fires when a stream is finished using a buffer.
+ *
+ * Once called, the buffer is no longer owned by the stream, and so may be
+ * reused or destroyed.
+ *
+ * This callback can run at any time, and from any thread; if you need to
+ * serialize access to your app's data, you should provide and use a mutex or
+ * other synchronization device.
+ *
+ * This function is extremely time-sensitive; the callback should do the least
+ * amount of work possible and return as quickly as it can. The longer the
+ * callback runs, the higher the risk of audio dropouts or other problems.
+ *
+ * \param userdata An opaque pointer provided by the app for their personal use.
+ * \param buffer The pointer to the buffer.
+ * \param buflen The length of the buffer.
+ * \sa SDL_PutAudioStreamBuffer
+ */
+typedef void (SDLCALL *SDL_ReleaseAudioBufferCallback)(void *userdata, const void *buffer, int buflen);
+
+/**
+ * Add data to the stream, without copying it.
+ *
+ * This data must match the format/channels/samplerate specified in the latest
+ * call to SDL_SetAudioStreamFormat, or the format specified when creating the
+ * stream if it hasn't been changed.
+ *
+ * Unlike SDL_PutAudioStreamData, this does not copy the data passed in.
+ * Instead, it takes ownership of the buffer.
```

Comment:
What does ownership of the buffer imply? What is the lifetime of the buffer? What happens when the stream is closed?